modules/apps/forms-and-workflow/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/display/context/CalendarDisplayContext.java

```diff
@@ -41,13 +47,30 @@ public CalendarDisplayContext(
 		_themeDisplay = themeDisplay;
 	}
 
-	public List<Calendar> getOtherCalendars(long[] calendarIds)
+	public List<Calendar> getOtherCalendars(User user, long[] calendarIds)
 		throws PortalException {
 
 		List<Calendar> otherCalendars = new ArrayList<>();
 
 		for (long calendarId : calendarIds) {
-			Calendar calendar = _calendarService.fetchCalendar(calendarId);
+			Calendar calendar = null;
+			try {
+				calendar = _calendarService.fetchCalendar(calendarId);
+			}
+			catch (PrincipalException principalException) {
+				if (_log.isInfoEnabled()) {
+					StringBundler sb = new StringBundler(4);
+
+					sb.append("No ");
+					sb.append(ActionKeys.VIEW);
+					sb.append(" permission for user ");
+					sb.append(user.getUserId());
+
+					_log.info(sb.toString(), principalException);
+				}
+
+				continue;
```

Comment:
is this line necessary ?

Reply:
@huynguyen-codeengine yes, even though we handle the exception. If we do not place this line here, it will go down to further operations that are not necessary.

Reply:
line 75 checks if the calendar is null it will continue, that is why we do not let it repeat itself.

Reply:
Calendar is null with exception so it does the same job with line 75 ?

Reply:
@huynguyen-codeengine thank you for your feedback, I removed the continue in line 72